### ISSUE TITLE
Errors parsing _vimrc and _vsvimrc files as of 2019-06-27

### ISSUE THREAD

**Issue by srhaymes56** [VsVim/VsVim#2662]
A few days ago, I started getting parsing errors in Visual Studio Community 2017 version 15.9.13 when opening both .py and .c files.  The parsing errors appear to be from either my _vimrc or _vsvimrc files as lines from these files are being echoed.

When uninstalling vsvim and then reinstalling from an **old** download of VsVim.vsix the errors temporarily go away until Visual Studio gives me a message that vsvim has been automatically updated and then the parsing errors return.  I went through 3 cycles of this.

I had to disconnect my network, uninstall vsvim, reinstall from the old VsVim.vsix file, disable "Automatically update this extension", and then reconnect my network.

My conclusion is that the latest version of vsvim that Visual Studio automatically installs is the source of the parsing errors.  Sorry, I cannot send you the errors as I did not copy them from the error window before installing a now stable and older version of vsvim.

Note that my _vimrc file has been used for many years in Vim and gVim.  The _vsvimrc just just removes the lines for my Vim plugins as Visual Studio will not use them.  I have tested the problem with vsvim set to use one of both of these files and all combinations result in errors. 

Thank you

**Comment by ricksladkey** [comment]
This is a duplicate of issue #2585 and the problem has already been resolved in PR #2589.

The problem occurs when you do not have a vsvim rc file but you do have a vim rc file. Previously VsVim would attempt to load your vim rc file, but would suppresses the reporting of any errors that you see. In VsVim 2.7.0, due to a bug, those errors (which always occurred), are now being reported.

To prevent these errors from being displayed you can:

1. Upgrade to the latest nightly build of VsVim (which includes a fix for that problem), or
2. You can create a vsvim rc file and only put in that file commands that VsVim understands.

**Comment by srhaymes56** [comment]
Hi Rick,
 I installed the latest nightly build VsVim_v2.8.237.vsix.  I am still getting these errors:

Error on line 7: Parse error
Error on line 8: Unknown option: compatible
Error on line 9: Unknown option: mousefocus
Error on line 10: Unknown option: columns
Error on line 12: Unknown option: colorcolumn
Error on line 14: Unknown option: ruler
Error on line 18: Unknown option: listchars
Error on line 25: Unknown option: shiftround
Error on line 29: Parse error
Error on line 30: Parse error
Error on line 31: Parse error
Error on line 32: Unknown option: showmatch
Error on line 33: Unknown option: foldmethod
Error on line 34: Unknown option: foldlevel 

So the latest nightly build did not fix this.  The line numbers are nice though.

Regards,

Steve

**Comment by ricksladkey** [comment]
@srhaymes56, unless @jaredpar disagrees, this is the new behavior for vsvimrc file: errors are reported so you can fix them. Since only VsVim looks at vsvimrc files, it seems reasonable that they should be checked for errors. This is compatible with how vim behaves as it reports errors in your vimrc file at startup.

There is a special exception for vimrc file errors when there is no vsvimrc file is because if you are using the same rc file for both vim and vsvim, you cannot get rid of the errors without affecting vim.

**Comment by srhaymes56** [comment]
Hi Rick,

First I would like to thank you for your work on this.  I understand what you are saving given how Vim reports errors.  So why is VsVim report errors on legitimate lines in my _vsvimrc file that Vim has not complained about for over 10 years that are in my _vimrc and _gvimrc files?  I removed all of the Vim plugin specific lines as they are not installed for Visual Studio.

Thanks,

Steve

Error on line 7: Parse error
Unknown option: compatible
Unknown option: mousefocus
Unknown option: columns
Unknown option: colorcolumn
Unknown option: ruler
Unknown option: listchars
Unknown option: shiftround
Error on line 29: Parse error
Error on line 30: Parse error
Error on line 31: Parse error
Unknown option: showmatch
Unknown option: foldmethod
Unknown option: foldlevel

**Comment by ricksladkey** [comment]
TL;DR: these errors were always there but they weren't previously reported.

In a perfect world, VsVim would implement every Vim feature, for example every setting. But in reality, Vim has many settings than VsVim hasn't implemented yet.

Take the `mousefocus` setting. This is a valid Vim setting, but not a valid VsVim setting because VsVim hasn't implemented `mousefocus` yet. This has always been the case, but now the error is actually reported.

It is critical that the user gets feedback for invalid settings, because it is very easy to make a typo with a regular setting like this:

```
set shifftwidth=4
```

The user meant to type `shiftwidth` but made a typo, and unless there is an error, will be mystified as to why shiftwidth isn't working as it is documented to work.

Anyway, there are many more ways than bad settings in which the attempt to load an rc file meant for vim into vsvim can fail.

**Comment by srhaymes56** [comment]
Hi Rick,

In a prefect world, there would be no mouse.  There would just be a keyboard.  Just kidding :-)

Overall, I like what VsVim has to offer and given that this is just a minor bug, I will keep using VsVim.  Thanks for the clarification and thanks for your help on this.

Steve
